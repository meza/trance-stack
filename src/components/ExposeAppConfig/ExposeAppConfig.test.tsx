import { expect, it } from 'vitest';
import { ExposeAppConfig } from './ExposeAppConfig';

describe('ExposeAppConfig', () => {
  const appConfig: AppConfig = {
    hotjarId: 'a-hotjar-id',
    googleAnalyticsId: 'ga-id',
    visitorId: 'a-visitor-id',
    isProduction: true,
    version: '0.0.0-dev',
    sentryDsn: 'a-sentry-dsn',
    posthogApi: 'a-posthog-api',
    posthogToken: 'a-posthog-token'
  };

  it('can expose the app config correctly', () => {
    // eslint-disable-next-line new-cap
    const markup = ExposeAppConfig({ appConfig: appConfig });
    expect(markup).toMatchSnapshot();
  });
});

