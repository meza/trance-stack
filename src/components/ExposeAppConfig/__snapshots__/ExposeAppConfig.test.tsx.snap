// Vitest Snapshot v1

exports[`ExposeAppConfig > can expose the app config correctly 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "window.appConfig = {\\"hotjarId\\":\\"a-hotjar-id\\",\\"googleAnalyticsId\\":\\"ga-id\\",\\"mixpanelToken\\":\\"a-mixpanel-token\\",\\"visitorId\\":\\"a-visitor-id\\",\\"isProduction\\":true,\\"mixpanelApi\\":\\"a-mixpanel-api\\",\\"splitToken\\":\\"a-split-token\\",\\"cookieYesToken\\":\\"a-cookieyes-token\\",\\"version\\":\\"0.0.0-dev\\",\\"sentryDsn\\":\\"a-sentry-dsn\\"}",
    }
  }
  id="app-config"
  suppressHydrationWarning={true}
/>
`;
